Compiling error introduced in ESPHome 2023.11.0 - "sh: 1: xtensa-esp32-elf-g++: not found"

### The problem

Compiling process fails with "sh: 1: xtensa-esp32-elf-g++: not found", Error 127.

Seems to have been introduced with ESPHome 2023.11.0; previous version compiled and uploaded fine.

### Which version of ESPHome has the issue?

2023.11.0

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2023.11.2

### What platform are you using?

ESP32

### Board

ESP32-DevKitC-VE

### Component causing the issue

_No response_

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

```txt
Updating /config/esphome/senshubproto.yaml
------------------------------------------------------------

INFO ESPHome 2023.11.0
INFO Reading configuration /config/esphome/senshubproto.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing senshubproto (board: esp32dev; framework: arduino; platform: platformio/espressif32@5.4.0)
--------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
Dependency Graph
|-- AsyncTCP-esphome @ 2.0.1
|-- WiFi @ 2.0.0
|-- FS @ 2.0.0
|-- Update @ 2.0.0
|-- ESPAsyncWebServer-esphome @ 3.1.0
|-- DNSServer @ 2.0.0
|-- ESPmDNS @ 2.0.0
|-- Wire @ 2.0.0
|-- SPI @ 2.0.0
|-- BSEC Software Library @ 1.6.1480
Compiling .pioenvs/senshubproto/src/esphome/components/adc/adc_sensor.cpp.o
sh: 1: xtensa-esp32-elf-g++: not found
*** [.pioenvs/senshubproto/src/esphome/components/adc/adc_sensor.cpp.o] Error 127
========================== [FAILED] Took 8.23 seconds ==========================
======== [ERROR] /config/esphome/senshubproto.yaml ========



======================== [SUMMARY] ========================
  - /config/esphome/senshubproto.yaml: FAILED
```


### Additional information

_No response_

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -68,7 +68,7 @@ ENV \
 # See: https://unix.stackexchange.com/questions/553743/correct-way-to-add-lib-ld-linux-so-3-in-debian
 RUN \
     if [ "$TARGETARCH$TARGETVARIANT" = "armv7" ]; then \
-        ln -s /lib/arm-linux-gnueabihf/ld-linux.so.3 /lib/ld-linux.so.3; \
+        ln -s /lib/arm-linux-gnueabihf/ld-linux-armhf.so.3 /lib/ld-linux.so.3; \
     fi
 
 RUN \

